openedx/core/djangoapps/user_api/accounts/views.py

```diff
@@ -202,3 +205,23 @@ def partial_update(self, request, username):
             )
 
         return Response(account_settings)
+
+
+class AccountDeactivationView(APIView):
+    """
+    Account deactivation viewset. Currently only supports POST requests.
+    Only admins can deactivate accounts.
+    """
+    permission_classes = (permissions.IsAuthenticated, IsStaffUser)
+
+    def post(self, request, username):
+        """
+        POST /api/user/v1/accounts/{username}/deactivate/
```

Comment:
Having `/deactivate/` in the path is not RESTful. A `POST` or `PUT` action against `/api/user/v1/accounts/{username}/` would be more standard. We might also use the `DELETE` action.